core/ipc/sink/kafka-impl/src/main/java/org/opennms/core/ipc/sink/kafka/KafkaRemoteMessageDispatcherFactory.java

```diff
@@ -73,16 +73,22 @@
     public <S extends Message, T extends Message> void dispatch(SinkModule<S, T> module, String topic, T message) {
         try (MDCCloseable mdc = Logging.withPrefixCloseable(MessageConsumerManager.LOG_PREFIX)) {
             LOG.trace("dispatch({}): sending message {}", topic, message);
-            final ProducerRecord<String,String> record = new ProducerRecord<>(topic, module.marshal(message));
-            try {
-                // From KafkaProducer's JavaDoc: The producer is thread safe and should generally be shared among all threads for best performance.
-                final Future<RecordMetadata> future = producer.send(record);
-                // The call to dispatch() is synchronous, so we block until the message was sent
-                future.get();
-            } catch (InterruptedException e) {
-                LOG.warn("Interrupted while sending message to topic {}.", topic, e);
-            } catch (ExecutionException e) {
-                LOG.error("Error occured while sending message to topic {}.", topic, e);
+
+            final ProducerRecord<String,byte[]> record = new ProducerRecord<>(topic, module.marshal(message));
+            // Keep sending record till it delivers successfully.
+            while(true) {
+                try {
+                    // From KafkaProducer's JavaDoc: The producer is thread safe and should generally be shared among all threads for best performance.
+                    final Future<RecordMetadata> future = producer.send(record);
+                    // The call to dispatch() is synchronous, so we block until the message was sent
+                    future.get();
+                    break;
+                } catch (InterruptedException e) {
+                    LOG.warn("Interrupted while sending message to topic {}.", topic, e);
+                    break;
```

Comment:
Propagate the interrupt in case anything else may need visibility of it.
```suggestion
                    Thread.currentThread().interrupt();
                    break;
```